iis-workflows/iis-workflows-ingest-pmc/src/main/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/PmcXmlHandler.java

```diff
@@ -190,10 +184,10 @@ public void endElement(String uri, String localName, String qName)
 			builder.getPages().setEnd(this.currentValue.toString().trim());
 			
 		} else if (hasAmongParents(qName, ELEM_AFFILIATION, this.parents, ELEM_ARTICLE_META)) {
-			CRFAffiliationParser affiliationParser = new CRFAffiliationParser();
-			String affStr = this.currentValue.toString();
-			if (affStr.trim().length()>0) {
-				try {
+			try {
+				CRFAffiliationParser affiliationParser = new CRFAffiliationParser();
+				String affStr = this.currentValue.toString();
+				if (affStr.trim().length()>0) {
```

Comment:
Agree, this is much cleaner. I have also reordered this condition check with `CRFAffiliationParser` instantiation.